src/upf_network.py

```diff
@@ -132,27 +231,18 @@ def set_mtu_size(self) -> None:
             logger.info("Setting MTU size to %s for interface %s", self.mtu_size, self.name)
             iface_record.set("mtu", self.mtu_size).commit()
             return
-        logger.warning("Interface %s not found in the network database", self.name)
-
-    def get_gateway_ip_address(self) -> str:
-        """Get the gateway IPv4 address of the given network interface."""
-        iface_index = self.get_index()
-        routes = self.ip_route.get_routes(family=AF_INET)
-        for route in routes:
-            oif = route.get_attr("RTA_OIF")
-            gateway_ip = route.get_attr("RTA_GATEWAY")
-            if oif == iface_index:
-                if gateway_ip:
-                    return gateway_ip
-        return ""
+        logger.warning(
+            "Setting MTU size for interface %s failed: Interface not found in the network database",  # noqa: E501
+            self.name,
+        )
 
     def get_index(self) -> int:
         """Get the index of the network interface."""
-        iface_index = self.ip_route.link_lookup(ifname=self.name)
-        if not iface_index:
+        try:
+            return self.network_db.interfaces[self.name].get("index")  # type: ignore[reportAttributeAccessIssue]  # noqa: E501
+        except KeyError:
```

Comment:
Why go for a KeyError instead of something like 

```
interface = self.network_db.interfaces.get(self.name)
if not interface:
    logger.warning("Interface %s not found", self.name)
    return -1
index = interface.get("index")
if not index:
    logger.warning("Interface index %s not found", index)
    return -1
```

This would bring more granularity and avoid ignoring static analysis warnings

Reply:
I'm not sure we need this granularity. If the interface exists it will always have an index. KeyError is thrown when there's no interface with a given name.
I'll look into static analysis warning, but personally I'd keep try-except.

Reply:
BTW, the code you proposed also generates a static analysis problem.